examples/platform/silabs/SiWx917/SiWxWifiInterface.cpp

```diff
@@ -843,44 +784,88 @@ void ProcessEvent(WfxEvent_t inEvent)
             {
                 /* Terminate with end of scan which is no ap sent back */
                 wifi_scan_configuration.type                   = SL_WIFI_SCAN_TYPE_ADV_SCAN;
-                wifi_scan_configuration.periodic_scan_interval = ADV_SCAN_PERIODICITY;
+                wifi_scan_configuration.periodic_scan_interval = kAdvScanPeriodicity;
             }
             else
             {
                 wifi_scan_configuration = default_wifi_scan_configuration;
             }
-            sl_wifi_set_scan_callback(bg_scan_callback_handler, NULL);
-            scan_results_complete = false;
+            sl_wifi_set_scan_callback(bg_scan_callback_handler, nullptr);
             wfx_rsi.dev_state |= WFX_RSI_ST_SCANSTARTED;
-            status = sl_wifi_start_scan(SL_WIFI_CLIENT_2_4GHZ_INTERFACE, NULL, &wifi_scan_configuration);
+
+            osSemaphoreAcquire(sScanInProgressSemaphore, osWaitForever);
+            status = sl_wifi_start_scan(SL_WIFI_CLIENT_2_4GHZ_INTERFACE, nullptr, &wifi_scan_configuration);
             if (SL_STATUS_IN_PROGRESS == status)
             {
-                osSemaphoreAcquire(sScanSemaphore, WIFI_SCAN_TIMEOUT_TICK);
+                osSemaphoreAcquire(sScanCompleteSemaphore, kWifiScanTimeoutTicks);
```

Comment:
Where is `sScanInProgressSemaphore` released in the case of `SL_STATUS_IN_PROGRESS == status`

Reply:
fixed